src/daft-local-execution/src/sinks/outer_hash_join_probe.rs

```diff
@@ -0,0 +1,417 @@
+use std::sync::Arc;
+
+use arrow2::bitmap::MutableBitmap;
+use common_error::DaftResult;
+use daft_core::{
+    prelude::{Schema, SchemaRef},
+    series::Series,
+};
+use daft_dsl::ExprRef;
+use daft_micropartition::MicroPartition;
+use daft_plan::JoinType;
+use daft_table::{GrowableTable, Probeable, Table};
+use indexmap::IndexSet;
+use tracing::{info_span, instrument};
+
+use super::streaming_sink::{StreamingSink, StreamingSinkOutput, StreamingSinkState};
+use crate::pipeline::PipelineResultType;
+
+struct IndexBitmapTracker {
+    bitmaps: Vec<MutableBitmap>,
```

Comment:
we should also think of the design where we have a single `Bitmap` for all rows since we know the exact number of rows.

Reply:
I did come up with a solution using prefix sums of the table lengths, but it doesn't play too well if we want to convert the Bitmaps -> BooleanArrays in order to do a mask_filter at the end.